core/federated-catalog-core/src/test/java/org/eclipse/edc/catalog/query/BatchedRequestFetcherTest.java

```diff
@@ -123,14 +137,29 @@ private CatalogRequestMessage createRequest() {
     }
 
     private Catalog emptyCatalog() {
-        return Catalog.Builder.newInstance().id("id").datasets(Collections.emptyList()).build();
+        return Catalog.Builder.newInstance().id("id").datasets(emptyList()).dataServices(emptyList()).build();
     }
 
     private Catalog createCatalog(int howManyOffers) {
         var datasets = IntStream.range(0, howManyOffers)
                 .mapToObj(i -> createDataset("dataset-" + i))
                 .collect(Collectors.toList());
 
-        return Catalog.Builder.newInstance().id("catalog").datasets(datasets).build();
+        var build = List.of(DataService.Builder.newInstance().build());
+        return Catalog.Builder.newInstance().id("catalog").datasets(datasets).dataServices(build).build();
+    }
+
+    // registers all the necessary transformers to avoid duplicating their behaviour in mocks
+    private void registerTransformers(JsonBuilderFactory factory, ObjectMapper mapper) {
```

Comment:
now this test deserves the `@ComponentTest` annotation

Reply:
it sure does. forgot that.